lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -0,0 +1,239 @@
+        let key: string;
+        // check state and remove associated cache
+        for (key in storage) {
+            if (!state || key.indexOf(state) !== -1) {
+                const splitKey = key.split(Constants.RESOURCE_DELIM);
+                const keyState = splitKey.length > 1 ? splitKey[splitKey.length-1]: null;
+                if (keyState === state && !this.tokenRenewalInProgress(keyState)) {
+                    this.removeItem(key);
+                    this.setItemCookie(key, "", -1);
+                    this.clearMsalCookie(state);
+                }
+            }
+        }
+        // delete the interaction status cache
+        this.removeItem(TemporaryCacheKeys.INTERACTION_STATUS);
+        this.removeItem(TemporaryCacheKeys.REDIRECT_REQUEST);
+    }
+
+    /**
+     * Return if the token renewal is still in progress
+     * @param stateValue
+     */
+    private tokenRenewalInProgress(stateValue: string): boolean {
+        const renewStatus = this.getItem(`${TemporaryCacheKeys.RENEW_STATUS}|${stateValue}`);
+        return !!(renewStatus && renewStatus === BrowserConstants.inProgress);
+    }
+
+    /**
+     * add value to cookies
+     * @param cName
+     * @param cValue
+     * @param expires
+     */
+    setItemCookie(cName: string, cValue: string, expires?: number): void {
+        let cookieStr = cName + "=" + cValue + ";";
+        if (expires) {
+            const expireTime = this.getCookieExpirationTime(expires);
+            cookieStr += "expires=" + expireTime + ";";
+        }
+
+        document.cookie = cookieStr;
+    }
+
+    /**
+     * get one item by key from cookies
+     * @param cName
+     */
+    getItemCookie(cName: string): string {
+        const name = cName + "=";
+        const ca = document.cookie.split(";");
+        for (let i = 0; i < ca.length; i++) {
+            let c = ca[i];
+            while (c.charAt(0) === " ") {
+                c = c.substring(1);
+            }
+            if (c.indexOf(name) === 0) {
+                return c.substring(name.length, c.length);
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Clear an item in the cookies by key
+     * @param cName
+     */
+    clearItemCookie(cName: string) {
+        this.setItemCookie(cName, "", -1);
+    }
+
+    /**
+     * Clear all msal cookies
+     */
+    public clearMsalCookie(state?: string): void {
+        const nonceKey = state ? `${TemporaryCacheKeys.NONCE_IDTOKEN}|${state}` : TemporaryCacheKeys.NONCE_IDTOKEN;
+        this.clearItemCookie(nonceKey);
+        this.clearItemCookie(TemporaryCacheKeys.REQUEST_STATE);
+        this.clearItemCookie(TemporaryCacheKeys.ORIGIN_URI);
+    }
+
+    /**
+     * Get cookie expiration time
+     * @param cookieLifeDays
+     */
+    getCookieExpirationTime(cookieLifeDays: number): string {
+        const today = new Date();
+        const expr = new Date(today.getTime() + cookieLifeDays * 24 * 60 * 60 * 1000);
```

Comment:
Can `24 * 60 * 60 * 1000` be made into a `const`?

Reply:
+1